include/zephyr/drivers/bluetooth/hci_driver.h

```diff
@@ -296,6 +296,19 @@ struct net_buf *bt_hci_cmd_complete_create(uint16_t op, uint8_t plen);
  */
 struct net_buf *bt_hci_cmd_status_create(uint16_t op, uint8_t status);
 
+#if defined(CONFIG_BT_BLUENRG_ACI)
+/** Set the BlueNRG public address.
+ *
+ * This function ensures the BlueNRG is configured using the specified public address. It needs to
+ * be called before bt_enable().
+ *
+ * @param addr     Public address to use.
+ *
+ * @return 0 on success, negative errno otherwise
+ */
+int bt_spi_bluenrg_set_pubaddr(bt_addr_t *addr);
+#endif /* CONFIG_BT_BLUENRG_ACI */
```

Comment:
`hci_driver.h` is not really the right place to put vendor/driver specific APIs.

This might be a good opportunity to add such a feature to Zephyr on a more generic level, by making it possible to pass a public address to the `bt_id_create()` API. Right now this API explicitly rejects anything that's not a static random address:

https://github.com/zephyrproject-rtos/zephyr/blob/2171f8b7a1770f5818fce0083175c8360004672c/subsys/bluetooth/host/id.c#L1284-L1289 

It's also worth noting that we've actually specified a Zephyr vendor HCI command that could be used for this, but so far I don't think any controller implements it (even Zephyr's own):
https://github.com/zephyrproject-rtos/zephyr/blob/2171f8b7a1770f5818fce0083175c8360004672c/doc/connectivity/bluetooth/api/hci.txt#L410-L415

What I'd propose for now, is to do the following:
 - Create a new HCI driver Kconfig option which adds a new callback to the HCI driver struct for setting the public address.
 - Update bt_id_create() to accept public addresses, and make it call the new driver callback (if available, otherwise return an error like right now)

Reply:
> * Update bt_id_create() to accept public addresses, and make it call the new driver callback (if available, otherwise return an error like right now)

Correction: you can't (and shouldn't) call the new "set public address" callback through `bt_id_create()` (that's only used to store the address), rather you need to modify the HCI init sequence to do this, most likely first thing in the `bt_init()` function.

Reply:
So where should I pass the public address from application to subsys? As an argument to `bt_enable()`->`init_work()`->`bt_init()`->`hci_init()`->some `struct bt_dev` method? That seems a bit roundabout.

Reply:
> So where should I pass the public address from application to subsys? As an argument to `bt_enable()`->`init_work()`->`bt_init()`->`hci_init()`->some `struct bt_dev` method? That seems a bit roundabout.

You would pass it to `bt_id_create()` before calling `bt_enable()`

Reply:
Just to clarify a bit more: `bt_id_create()` would store the address in `bt_dev.id_addr[0]` and the HCI init sequence would pick it from there (the init sequence would check for the address type and if it's public it would call the new HCI driver callback)

Reply:
Ahh I see, that makes sense, thanks for the explanation.

Reply:
Updated as requested.